File: sem3/map/ToyLanguageInterpreter/src/view/commands/Command.java
package view.commands;

import utils.exceptions.InterpreterException;

import java.util.Scanner;

/**
 * Created by mirko on 14/11/2016.
 */
public abstract class Command {
    private String key;
    private String description;

    public Command(String key, String description) {
        this.key = key;
        this.description = description;
    }

    public abstract void execute(Scanner scanner) throws InterpreterException;

    public String getKey() {
        return key;
    }

    public String getDescription() {
        return description;
    }
}

File: sem3/map/ToySharpLanguageInterpreter/ToySharpLanguageInterpreter/src/view/commands/Command.cs
using System;
using System.IO;

namespace ToySharpLanguageInterpreter
{
	public abstract class Command
	{
		private string key;
		private string description;

		public Command(string key, string description)
		{
			this.key = key;
			this.description = description;
		}

		public abstract void execute();

		public string getKey()
		{
			return key;
		}

		public string getDescription()
		{
			return description;
		}
	}
}